apps/system-e2e/src/support/utils.ts

```diff
@@ -13,4 +14,55 @@ export const getCognitoCredentials = () => {
   }
 }
 
+export const hasOperationName = (
+  req: CyHttpMessages.IncomingHttpRequest,
+  operationName: string,
+) => {
+  const { body } = req
+  return typeof body === 'object' && body.operationName === operationName
+}
+
+export const aliasQuery = (
+  req: CyHttpMessages.IncomingHttpRequest,
+  operationName: string,
+) => {
+  if (hasOperationName(req, operationName)) {
+    req.alias = operationName
+  }
+}
+
+export const getFakeUser = (fakeUsers: FakeUser[], name: string) =>
+  fakeUsers
+    .filter((e) => e.name.toLowerCase().includes(name.toLowerCase()))
+    .reduce((e) => e)
+
+export const getFamily = (user: FakeUser) => {
+  return [
+    Object.assign(
+      {},
+      ...Object.entries(user)
+        .filter(([k]) => k !== 'children')
+        .map(([k, v]) => ({ [k]: v })),
+    ),
+    ...user.children,
+  ]
+}
+
+const getDiscountUser = (fakeUser: FakeUser, discounts: Discount[]) =>
+  discounts.filter((e) => e.user.name === fakeUser.name).reduce((e) => e)
```

Comment:
The reduce looks strange

Reply:
To return the user as object instead of pop-ing the array